pre_commit/resources/pre-push-hook

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+# This is a randomish md5 to identify this script
+# 49fd668cb42069aa1b6048464be5d395
+
+pushd `dirname $0` > /dev/null
+HERE=`pwd`
+popd > /dev/null
+
+retv=0
+
+ENV_PYTHON='{sys_executable}'
+
+which pre-commit >& /dev/null
+WHICH_RETV=$?
+"$ENV_PYTHON" -c 'import pre_commit.main' >& /dev/null
+ENV_PYTHON_RETV=$?
+python -c 'import pre_commit.main' >& /dev/null
+PYTHON_RETV=$?
+
+
+if ((
+            (WHICH_RETV != 0) &&
+            (ENV_PYTHON_RETV != 0) &&
+            (PYTHON_RETV != 0)
+            )); then
```

Comment:
this doesn't match `pre-commit-hook`, I wonder if there's a way we could combine the duplicated code somehow